src/plugins/dataflow-tool/nodes/factories/control-rete-node-factory.tsx

```diff
@@ -0,0 +1,114 @@
+import Rete, { Node, Socket } from "rete";
+import { NodeData } from "rete/types/core/data";
+import { DataflowReteNodeFactory, kEmptyValueString } from "./dataflow-rete-node-factory";
+import { ValueControl } from "../controls/value-control";
+import { DropdownListControl } from "../controls/dropdown-list-control";
+import { NodeControlTypes, roundNodeValue } from "../../model/utilities/node";
+import { PlotButtonControl } from "../controls/plot-button-control";
+
+export class ControlReteNodeFactory extends DataflowReteNodeFactory {
+  constructor(numSocket: Socket) {
+    super("Control", numSocket);
+  }
+
+  private heldValue: number | null = null;
+  private result = 0;
+  private resultSentence = "";
+
+  public builder(node: Node) {
+    super.defaultBuilder(node);
+    if (this.editor) {
+      const inp1 = new Rete.Input("num1", "Binary", this.numSocket);
+      const inp2 = new Rete.Input("num2", "Number2", this.numSocket);
+      const out = new Rete.Output("num", "Number", this.numSocket);
+
+      node.data.hasGate = true;
+      node.data.gateActive = false;
+
+      const dropdownOptions = NodeControlTypes
+        .map((nodeOp) => {
+          return { name: nodeOp.name, icon: nodeOp.icon };
+        });
+      return node
+        .addInput(inp1)
+        .addInput(inp2)
+        .addControl(new DropdownListControl(this.editor, "controlOperator", node, dropdownOptions, true))
+        .addControl(new PlotButtonControl(this.editor, "plot", node))
+        .addControl(new ValueControl(this.editor, "nodeValue", node))
+        .addOutput(out) as any;
+    }
+  }
+
+  private setHoldModValue(n1:number, n2: number, prior: number, funcName: string) :number {
+    if (funcName === "Output Zero"){
+      this.heldValue = null;
+      return 0;
+    }
+
+    else if (funcName === "Hold Current"){
+      if (this.heldValue === null){
+        this.heldValue = n2;
+      }
+      return this.heldValue;
+    }
+
+    else if (funcName === "Hold Prior"){
+      if (this.heldValue === null){
+        this.heldValue = prior;
+      }
+      return this.heldValue;
+    }
+
+    else {
+      return 0;
+    }
+  }
+
+  public worker(node: NodeData, inputs: any, outputs: any) {
+    const funcName = node.data.controlOperator as string;
+    const recents: number[] = (node.data.recentValues as any).nodeValue;
+    const priorValue: number | undefined = recents[recents.length - 1];
```

Comment:
Instead of looking at the `recentValues` for the prior, you could use a private class member that stores the current value at the end of each call to `worker()`.

Reply:
I see.  Is there a reason that's more performant or clearer? I did think about the "delay" block idea down the road and so thought this was a start to accessing those prior values.

Reply:
I don't think there's a big disadvantage to using `recentValues`. That said, I'm not sure `recentValues` would actually work for the delay block, because `recentValues` would continue to tick while that block is holding. Plus, who knows if the delay block will ever see the light of day :D